app/src/main/java/com/dodo/flashcards/presentation/MainActivity.kt

```diff
@@ -11,39 +12,60 @@ import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavHostController
+import com.dodo.flashcards.architecture.Router
 import com.dodo.flashcards.presentation.register_screen.RegisterScreen
 import com.dodo.flashcards.presentation.theme.FlashcardsAppTheme
+import com.dodo.flashcards.util.Screen.*
 import dagger.hilt.android.AndroidEntryPoint
+import com.dodo.flashcards.presentation.MainDestination.*
+import com.google.accompanist.navigation.animation.rememberAnimatedNavController
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
 
 @AndroidEntryPoint
-class MainActivity : ComponentActivity() {
+class MainActivity : ComponentActivity(), Router<MainDestination> {
+
+    lateinit var navController: NavHostController
+    private lateinit var auth: FirebaseAuth
+
+    @OptIn(ExperimentalAnimationApi::class)
     override fun onCreate(savedInstanceState: Bundle?) {
+
+        auth = Firebase.auth
+
         super.onCreate(savedInstanceState)
+
         setContent {
             FlashcardsAppTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colors.background
-                ) {
-                    // Todo, we need to define router and move this to the NavHost
-                    // This is temporary to test registration
-                    RegisterScreen(viewModel = hiltViewModel())
-                }
+                navController = rememberAnimatedNavController()
+                val currentUser = auth.currentUser
+                val startScreen = if (currentUser != null) Welcome.route else Register.route
```

Comment:
This is temporary so fine for now, but will probably actually have the else be the login screen rather than register